go/mysql/auth_server.go

```diff
@@ -117,8 +119,8 @@ func NewSalt() ([]byte, error) {
 	return salt, nil
 }
 
-// ScramblePassword computes the hash of the password using 4.1+ method.
```

Comment:
We are changing the interface here. If there is someone out there who built their own plugin auth server, this will break them.
We need to (at least) document this in the 10.0 release notes.

Reply:
I do not see this as interface. If you think it can break someone's build, we can mark this as breaking change.

Reply:
You are correct, it is not part of the AuthServer interface. My mistake.